pkg/client/typed/dynamic/client.go

```diff
@@ -79,28 +84,54 @@ func NewClient(conf *restclient.Config) (*Client, error) {
 	return &Client{cl: cl}, nil
 }
 
-// Resource returns an API interface to the specified resource for
-// this client's group and version. If resource is not a namespaced
-// resource, then namespace is ignored.
+// Resource returns an API interface to the specified resource for this client's
+// group and version. If resource is not a namespaced resource, then namespace
+// is ignored.
 func (c *Client) Resource(resource *unversioned.APIResource, namespace string) *ResourceClient {
-	return &ResourceClient{
+	rc := &ResourceClient{
 		cl:       c.cl,
 		resource: resource,
 		ns:       namespace,
 	}
+	return rc
+}
+
+// ParameterCodec wraps a parameterCodec around the Client.
+func (c *Client) ParameterCodec(parameterCodec runtime.ParameterCodec) *ClientWithParameterCodec {
```

Comment:
Maybe for 1.3 the earlier change of passing an optional (nil or not)
parameter codec moves us in the right direction.  There's probably a lot of
changes we can make to dynamic client as it is used in more places (and we
start needing to worry about its performance) - I'd be ok with a TODO and
follow up in 1.4



On Jun 3, 2016, at 1:29 PM, Chao Xu <notifications@github.com> wrote:

In pkg/client/typed/dynamic/client.go
<https://github.com/kubernetes/kubernetes/pull/26493#discussion_r65745313>:

>  		cl:       c.cl,
>  		resource: resource,
>  		ns:       namespace,
>  	}
> +	return rc
> +}
> +
> +// ParameterCodec wraps a parameterCodec around the Client.
> +func (c *Client) ParameterCodec(parameterCodec runtime.ParameterCodec) *ClientWithParameterCodec {

How about using a wrapper?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/26493/files/9e63bbb93387ea03634786063bfdf45197aaf47b#r65745313>,
or mute the thread
<https://github.com/notifications/unsubscribe/ABG_p18MkMzsOPWtZ7ZA_qXFVuIVfhNlks5qIGSLgaJpZM4IpLXf>
.